src/common/components/ResponsiveImage/index.tsx

```diff
@@ -1,16 +1,88 @@
-import React, { ImgHTMLAttributes } from 'react';
+import React, { FC, ImgHTMLAttributes, useEffect, useState } from 'react';
+
+import { useBoundingRect } from 'common/hooks/useBoundingRect';
+import { getKeys } from 'common/utils/tsHacks';
+
 import { DOMAIN } from '../../constants/endpoints';
-import IResponsiveImage, { IResponsiveImageSizes } from '../../models/ResponsiveImage';
+import IResponsiveImage, {
+  IMAGE_DIMENTIONS,
+  IResponsiveImageSizes,
+  ResponsiveImageTypes,
+} from '../../models/ResponsiveImage';
+
+export type ImageType = ResponsiveImageTypes;
+export type ImageSizes = IResponsiveImageSizes;
+export type ImageSize = keyof ImageSizes;
 
 export interface IProps extends ImgHTMLAttributes<HTMLImageElement> {
   image: IResponsiveImage;
-  size: keyof IResponsiveImageSizes;
+  size: ImageSize;
+  type: ImageType;
 }
 
-export const ResponsiveImage = ({ image, size, alt, ...props }: IProps) => {
-  const defaultImage = image[size];
+/**
+ * @summary Get the key types of images sorted by width.
+ * @example ['xs', 'sm', 'md', 'lg']
+ * @param {ImageType} type
+ */
+const getSortedDimentionKeys = (type: ImageType) => {
+  return getKeys(IMAGE_DIMENTIONS[type])
+    .sort((a, b) => {
+      const [aWidth] = IMAGE_DIMENTIONS[type][a];
+      const [bWidth] = IMAGE_DIMENTIONS[type][b];
+      return aWidth - bWidth;
+    })
+    .filter((t) => t !== 'wide'); // Filter wide because it is unstable, and just mimicks 'lg'.
+};
+
+/**
+ * @summary Get the most fitting image version to display on screen based on the client screen/pixel size.
+ * @param {ResponsiveImageTypes} type
+ * @param {number} displayWidth The actual width of the displayed image in real pixel in the screen.
+ * @param {ImageSize} currentSize The default/currently displayed size. Resolution should ne be downgraded.
```

Comment:
Typo?

Reply:
You are correct. I probably need better linting on my spelling 🤷‍♂️